flang/docs/ParallelFortranRuntime.md

```diff
@@ -0,0 +1,1987 @@
+  after [`prif_change_team`](#prif_change_team), or to pass to a coarray dummy
+  argument (especially in the case that the cobounds are different)
+    
+```
+subroutine prif_alias_create( &
+    source_handle, alias_lcobounds, alias_ucobounds, alias_handle)
+  type(prif_coarray_handle), intent(in) :: source_handle
+  integer(c_intmax_t), intent(in) :: alias_lcobounds(:)
+  integer(c_intmax_t), intent(in) :: alias_ucobounds(:)
+  type(prif_coarray_handle), intent(out) :: alias_handle
+end subroutine
+```
+**Further argument descriptions**:
+
+* **`source_handle`**: a handle (which may itself be an alias) to the existing
+  coarray for which an alias is to be created
+* **`alias_lcobounds` and `alias_ucobounds`**: the cobounds to be used for
+  the new alias. Both arguments must have the same size, but it need not
+  match the corank associated with `source_handle`
+* **`alias_handle`**: a new alias to the existing coarray
+
+### `prif_alias_destroy`
+
+**Description**: Delete an alias to a coarray. Does not deallocate the original coarray.
+    
+```
+subroutine prif_alias_destroy(alias_handle)
+  type(prif_coarray_handle), intent(in) :: alias_handle
+end subroutine
+```
+**Further argument descriptions**:
+
+* **`alias_handle`**: the alias to be destroyed
+
+### `MOVE_ALLOC`
+
+This is not provided by PRIF because it depends on unspecified details
+of the compiler's `allocatable` attribute. It is the compiler's responsibility
+to implement `MOVE_ALLOC` using PRIF-provided operations. For example, according
+to the Fortran standard, `MOVE_ALLOC` with coarray arguments is an image control statement that
+requires synchronization, so the compiler should likely insert call(s) to
+`prif_sync_all` as part of the implementation.
+
+| **CLIENT NOTE**: |
+| ---------------- |
+| It is envisioned that the use of `prif_set_context_data` and `prif_get_context_data` will allow for an efficient implementation of `MOVE_ALLOC` that maintains tracking of allocation status |
+
+## Coarray Queries
+
+### `prif_set_context_data`
+
+**Description**: This procedure stores a `c_ptr` associated with a coarray
+  for future retrieval. A typical usage would be to store a reference
+  to the actual variable whose allocation status must be changed in the case
+  that the coarray is deallocated.
+    
+```
+subroutine prif_set_context_data(coarray_handle, context_data)
+  type(prif_coarray_handle), intent(in) :: coarray_handle
+  type(c_ptr), intent(in) :: context_data
+end subroutine
+```
+
+### `prif_get_context_data`
+
+**Description**: This procedure returns the `c_ptr` provided in the most
+  recent call to [`prif_set_context_data`](#prif_set_context_data) with the
+  same coarray (possibly via an alias coarray handle).
+    
+```
+subroutine prif_get_context_data(coarray_handle, context_data)
+  type(prif_coarray_handle), intent(in) :: coarray_handle
+  type(c_ptr), intent(out) :: context_data
+end subroutine
+```
+
+### `prif_base_pointer`
```

Comment:
Quoted from @jeffhammond's response:

> `prif_base_pointer` should be immediately prior to the raw operations since the usage will be paired. I know it's a query but locality of reference when reading the document is important.

Reply:
**LBL response:** `prif_base_pointer`  is currently a mandatory prerequisite for calls in many different API sections. We will add a forward reference to those sections that “consume” the result of this coarray query. However based on your feedback we are considering a new formulation where `prif_base_pointer` would either only be needed before a call affecting indirect data or not needed at all.

Reply:
In v0.4, `prif_base_pointer` has been removed, as it has been made redundant by new direct versions of many procedures.